### PR TITLE
Add a new submodule libuv v1.34.0

### PR DESCRIPTION
**Pull request by guantaol**

### CODE REVIEW

**Comment by jtattermusch** [comment]
The description should mention that we're using the v1.34.0 release (nonobvious from just the commit sha).
https://github.com/libuv/libuv/releases/tag/v1.34.0

**Comment by jtattermusch** [comment]
Several sanity tests are failing
https://source.cloud.google.com/results/invocations/437d5acc-d2af-4f4a-bc3b-74e3b5e1e4ac

File: bazel/grpc_deps.bzl [hunk 1]
```diff
@@ -263,6 +268,15 @@ def grpc_deps():
             sha256 = "122ebf7fe7d1c8e938af6aeaee0efe788a3a2449ece5a8d6a428cb18d6f88033",
         )
 
+    if "libuv" not in native.existing_rules():
+       http_archive(
+           name = "zlib",
```
Comment (gnossen) [inline bazel/grpc_deps.bzl]: `name = "libuv"`?

Comment (guantaol) [reply bazel/grpc_deps.bzl]: Thanks for catching this!

**Comment by guantaol** [comment]
Sorry for the confusion! This PR is still a draft, and I was intended to share it only with Nico. Do not know why the other reviewers are auto added by github.

**Comment by jtattermusch** [comment]
> Sorry for the confusion! This PR is still a draft, and I was intended to share it only with Nico. Do not know why the other reviewers are auto added by github.

They are added automatically by github because they are listed as OWNERs for some portions of code this PR is touching.

**Comment by guantaol** [comment]
Known issues:

Bazel C/C++ Dbg MacOS - #20814
Basic Tests C/C++ MacOS - #21534

**Pull request merged**